### ISSUE TITLE
Freedom issue: music requires proprietary software to compile

### ISSUE THREAD

**Issue by throwaway1037** [singularity/singularity#322]
Please see <https://labs.parabola.nu/issues/3233>.

To solve these issues, I propose the switch to a cross-platform, libre music program and libre instrument data.

Thank you.

**Comment by rofl0r** [comment]
i dont think it matters which programs were used to "compile" the music, as long as the end result is available under a permissive license. is there even input data to recreate the music using the listed tools ?

**Comment by throwaway1037** [comment]
The music is proprietary if its complete and corresponding source form is not published, libre, and usable with free software. Merely publishing the end result would be analogous to publishing an executable with no corresponding source code.

**Comment by rofl0r** [comment]
> The music is proprietary if its complete and corresponding source form is not published

that sounds quite nonsenical to me. most artwork (images, etc) on e.g. wikipedia are CC0 but nobody documented how and using which tools they created svgs, png, jpegs, video, and audio samples published there, yet i've never heard anyone claiming these data are proprietary and non-free.

**Comment by nomadfox** [comment]
> that sounds quite nonsensical to me. most artwork (images, etc) on e.g. wikipedia are CC0 but nobody documented how and using which tools they created svgs, png, jpegs, video, and audio samples published there, yet i've never heard anyone claiming these data are proprietary and non-free.

Absolutely right, if you are going to judge whether the work is libre or not by which tools were used to compile it, then you'd have to take into account not only the final tool like editing software but also things like the operating system, the issue tracking tools and any other non libre software that was used in the process, that would make a lot of already released work "non-libre".

**Comment by throwaway1037** [comment]
The issue is not which tools were used to invent the work, but whether the work is practically modifiable in its final form. For instance, pixel art is generally modifiable once exported to a libre image format because an image editor can read the file and write changes to it easily. However, consider the case that the final image was generated from a 3D model. In that case, the work is sufficiently complex that the source form of the work (the 3D model and 3D software project file, eg. Blender `.blend` file) would be required to make substantial changes, such as rotating the model in 3D space, or changing the textures and bump maps, and then re-rendering another final image. Providing a prerendered image of such a scene which is free in itself would not allow such changes, which is analogous to the case of providing a free executable without its complete and corresponding source code; without the latter, the former is rendered nonfree.

This case is an instance of providing the precompiled final form of a work (the analogy of "object code"), without the complete and corresponding source form also being libre; it requires proprietary instrument data. Moreover, it requires proprietary software.

**Comment by nomadfox** [comment]
I get your point, it would be difficult to collaborate on work that was made using proprietary software that not all participants can use, but this does not change "the rights" each participant has on the work, they would still be **legally** able to modify, recompile, redistribute... the work, therefore the work itself is still free.
To me this is more of a collaboration convenience issue than freedom issue, something to keep in mind rather than require.

**Comment by throwaway1037** [comment]
It's equivalent to the case in which the final executable is free, but the source code requires a proprietary module and a proprietary compiler to be built, thus rendering the work as a whole nonfree because it can't be worked with properly in the free world.

May I propose switching to a libre, cross-platform, maintained audio program and instrument pack? This would also have the practical benefit as you mention of improving the music's ability to be used in collaboration with other contributors.

**Comment by throwaway1037** [comment]
I was under the impression that `data/music-src` held the partial sources for the music aside from the required proprietary instruments and software in the form of Drumatic VE VST and Renoise.

**Comment by rofl0r** [comment]
> Unfortunately, I'm not much of a musician and I'm not familiar with any DAWs, so I can't do the work unless I spend time learning how to use one.

i have serious doubts anyone else is interested in doing this work for an ideological change, but you never know.

**Comment by rofl0r** [comment]
> There are the practical benefits I mentioned earlier

yeah, assuming someone actually wants to change the music, which in my opinion is pretty good already.

**Comment by throwaway1037** [comment]
Or wants to compose new music, or remix the existing music; this would also help long-term sustainability.
